File: Python3/프로그래머스/3/42884. 단속카메라/단속카메라.py
def solution(routes):
    cams, flag = [], True
    
    for route in sorted(routes, key=lambda x:x[0]):
        flag = False
        for i in range(len(cams)):
            if route[0] <= cams[i][1]:
                # print(route, "is in", cams[i])
                flag = True
                cams[i][0],cams[i][1] = route[0], min(route[1], cams[i][1])
        if not flag:
            cams.append(route)
        # print("CAMS:", cams)
    
    return len(cams)

File: C++/프로그래머스/3/42884. 단속카메라/단속카메라.cpp
#include <vector>
#include <algorithm>

using namespace std;

int solution(vector<vector<int>> routes) {
    vector<vector<int>> cams;
    bool flag;
    sort(routes.begin(), routes.end(), [&](vector<int> a, vector<int> b)->bool {
        return a[0] < b[0];
    });
    
    for(auto& route : routes) {
        flag = false;
        for(int i = 0; i < cams.size(); i++) {
            if(route[0] <= cams[i][1]) {
                flag = true;
                cams[i][0] = max(route[0], cams[i][0]);
                cams[i][1] = min(route[1], cams[i][1]);
            }
        }
        if(!flag) {
            cams.push_back(route);
        }
    }
    
    
    return cams.size();
}